### PR TITLE
GH-14 Configure basic Redux and userSlice

### PR DESCRIPTION
**Pull request by mskwara**
Let me know if something is missing or should be done in another way. This is the first time I used Redux :(

### CODE REVIEW

File: src/providers/redux/userSlice.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const userSlice = createSlice({
+  name: 'user',
+  initialState,
+  reducers: {
+    updateUser: (state, action: { payload: IUserState }) => {
+      const { id, name, surname, email } = action.payload;
+      state.id = id;
+      state.name = name;
+      state.surname = surname;
+      state.email = email;
```
Comment (mskwara) [inline src/providers/redux/userSlice.ts]: Is there a better way to assign these values?

Comment (jakobos) [reply src/providers/redux/userSlice.ts]: Maybe :
`
state = {
id: id,
name: name,
surname: surname,
email: email,
}
`

Comment (mskwara) [reply src/providers/redux/userSlice.ts]: It does not work for some reason

File: src/@types/redux.d.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export interface IUserState {
```
Comment (jakobos) [inline src/@types/redux.d.ts]: there is no need to create .d.ts file, there is no declaration, only interface IUserState, so you could delete this file and move UserState type to userSlice.ts

Maybe also you could rename it to 'UserState' I can't understand why we need the 'I' suffix before interface name :)

File: src/hooks/useUserData.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { IUserState } from '~/@types/redux';
+import { RootState } from '~/providers/redux/store';
+import { clearUser, updateUser } from '~/providers/redux/userSlice';
+
+const useUserData = () => {
+  const user = useSelector((state: RootState) => state.user);
```
Comment (jakobos) [inline src/hooks/useUserData.ts]: please add an arrow function `userSelector = (state: RootState) => state.user` in `userSlice.ts` and use it as argument for `useSelector` hook

File: src/providers/redux/userSlice.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  name: '',
+  surname: '',
+  email: '',
+};
+
+const userSlice = createSlice({
+  name: 'user',
+  initialState,
+  reducers: {
+    updateUser: (state, action: { payload: IUserState }) => {
```
Comment (jakobos) [inline src/providers/redux/userSlice.ts]: maybe we could add `UpdateUserAction` type and use it with `action: UpdateUserAction` ?

Comment (mskwara) [reply src/providers/redux/userSlice.ts]: I created interface:
```
export interface ActionPayload<T> {
  payload: T;
}
```
to be easily used with other reducers and changed line above to:
`action: ActionPayload<UserState>`

File: src/hooks/useUserData.ts (hunk 3 above)
Comment (mskwara) [reply src/hooks/useUserData.ts]: Ok, done

File: src/@types/redux.d.ts (hunk 2 above)
Comment (mskwara) [reply src/@types/redux.d.ts]: ✔️
I thought that I-prefix is a interface naming convention, but I searched the internet and found many statements that I-prefix shouldn't be used.
So I changed it.

File: src/providers/redux/userSlice.ts (hunk 1 above)
Comment (mskwara) [reply src/providers/redux/userSlice.ts]: Bartek suggested to just return the new state in this function and it worked.
```
updateUser: (state, action: ActionPayload<UserState>) => ({
  ...state,
  ...action.payload,
}),
```

**Review by jakobos — DISMISSED** [review]

File: src/providers/redux/store.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { configureStore } from '@reduxjs/toolkit';
+import userReducer from './userSlice';
+
+const store = configureStore({
+  reducer: {
+    user: userReducer,
+  },
+});
+
+export interface ActionPayload<T> {
```
Comment (bartoszskwara) [inline src/providers/redux/store.ts:10]: Please use import { PayloadAction } from '@reduxjs/toolkit';

File: src/providers/redux/userSlice.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  name: '',
+  surname: '',
+  email: '',
+};
+
+const userSlice = createSlice({
+  name: 'user',
+  initialState,
+  reducers: {
+    updateUser: (state, action: ActionPayload<UserState>) => ({
```
Comment (bartoszskwara) [inline src/providers/redux/userSlice.ts]: Please use import { PayloadAction } from '@reduxjs/toolkit';

**Review by bartoszskwara — DISMISSED** [review]

File: src/providers/redux/userSlice.ts (hunk 6 above)
Comment (mskwara) [reply src/providers/redux/userSlice.ts]: Ok

**Pull request merged**